### PR TITLE
Add GM7 and GM7+up as cloudsource

### PR DESCRIPTION
**Pull request by vmoravec**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -923,7 +923,7 @@ function onadmin_prepare_cloud_repos
             addcloud6pool
             addcloud6maintupdates
             ;;
-        develcloud7|susecloud7|M?|Beta*|RC*|GMC*)
+        develcloud7|susecloud7|GM7|GM7+up|M?|Beta*|RC*|GMC*)
             addcloud7pool
             addcloud7maintupdates
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: when looking above, the difference between GM and GM+up is that the latter has maintenance update repos - we should keep that distinction here, too

**Comment by rsalevsky** [comment]
I guess we can remove the GMC support?

**Comment by bmwiedemann** [comment]
There are no GMCs on dist anymore, but we should just move those with Beta and M* to cloud8 and make cloud7 be similar to 5 and 6

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -430,7 +430,7 @@ function add_sdk_repo
             zypper ar -p $sdk_repo_priority -f http://$susedownload/update/build.suse.de/SUSE/Products/SLE-SDK/12-SP1/$arch/product/ SDK-SP1
             zypper ar -p $sdk_repo_priority -f http://$susedownload/update/build.suse.de/SUSE/Updates/SLE-SDK/12-SP1/$arch/update/ SDK-SP1-Update
             ;;
-        develcloud7|mitakacloud7|susecloud7|M?|Beta*|RC*|GMC*)
+        develcloud7|mitakacloud7|susecloud7|GM7|GM7+up|M?|Beta*|RC*|GMC*)
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: susecloud7 should not exist anymore, can you remove it?

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1204,15 +1210,16 @@ function onadmin_set_source_variables
...
+            [[ $cs =~ GM7 ]] && cs=GM
             CLOUDSLE12DISTPATH=/install/SLE-12-SP2-Cloud7-$cs/
             CLOUDSLE12DISTISO="SUSE-OPENSTACK-CLOUD-7-$arch*1.iso"
-            CLOUDSLE12TESTISO="CLOUD-7-TESTING-$arch*Media1.iso"
+            CLOUDSLE12TESTISO="CLOUD-7-TESTING-$arch*DVD1.iso"
             CLOUDLOCALREPOS="SUSE-OpenStack-Cloud-7-official"
         ;;
         *)
-            complain 76 "You must set environment variable cloudsource=develcloud4|develcloud5|develcloud6|develcloud7|susecloud7|GM4+up|GM5|Mx|GM6"
+            complain 76 "You must set environment variable cloudsource=develcloud4|develcloud5|develcloud6|develcloud7|susecloud7|GM4+up|GM5|Mx|GM6|GM7"
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: Same here, remove susecloud7 or replace it with susecloud8

**Comment by vmoravec** [comment]
@dirkmueller @bmwiedemann @rsalevsky Addressed all your comments

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -961,7 +967,7 @@ function onadmin_prepare_cloud_repos
             develcloud6)
                 addsles12sp1testupdates
                 ;;
-            develcloud7|mitakacloud7|susecloud7|M?|Beta*|RC*|GMC*)
+            develcloud7|mitakacloud7|GM7|GM7+up|M?|Beta*|RC*|GMC*)
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: better copy the GM6 / GM6+up parts above?

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -1204,15 +1210,16 @@ function onadmin_set_source_variables
             CLOUDSLE12DISTISO="SUSE-OPENSTACK-CLOUD-6-$arch*1.iso"
             CLOUDLOCALREPOS="SUSE-OpenStack-Cloud-6-official"
         ;;
-        GMC*|M?)
+        GM7|GM7+up|GMC*|M?)
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: no GMC/M* for cloud7 anymore -> move those to 8

File: scripts/qa_crowbarsetup.sh [hunk 6]
```diff
@@ -4116,7 +4123,7 @@ function oncontroller_prepare_functional_tests
 {
     if iscloudver 6plus; then
         case "$cloudsource" in
-            develcloud?|susecloud?|M?|Beta*|RC*|GMC*)
+            develcloud?|susecloud?|M?|Beta*|RC*|GMC*|GM*)
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: what is this case/match meant to filter? it seems to cover everything...

File: scripts/qa_crowbarsetup.sh (hunk 5 above)
Comment (vmoravec) [inline scripts/qa_crowbarsetup.sh]: In this case I would like to keep it here as the code right below this line shows what to do when cloud8 is available.

Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: just add it now - then it is already there when it is available

**Comment by vmoravec** [comment]
Closing in favor of https://github.com/SUSE-Cloud/automation/pull/1736

**Pull request closed without merging**